MNE/disp3D/3DObjects/brain/brainsurfacetreeitem.cpp

```diff
@@ -221,14 +229,48 @@ bool BrainSurfaceTreeItem::addData(const Surface& tSurface, Qt3DCore::QEntity* p
 
 //*************************************************************************************************************
 
+void BrainSurfaceTreeItem::onRtVertColorChanged(const QByteArray& sourceColorSamples)
+{
+    //Set new data.
+    //In setData(data, BrainSurfaceTreeItemRoles::SurfaceCurrentColorVert) we pass the new color values to the renderer (see setData function).
+    QVariant data;
+    data.setValue(sourceColorSamples);
+    this->setData(data, BrainSurfaceTreeItemRoles::SurfaceRTSourceLocColor);
+    this->setData(data, BrainSurfaceTreeItemRoles::SurfaceCurrentColorVert);
+}
+
+
+//*************************************************************************************************************
+
+void BrainSurfaceTreeItem::onAnnotationVisibilityChanged(bool isVisible)
+{
+    if(isVisible) {
+        m_sColorInfoOrigin = "Color from annotation";
+    } else {
+        m_sColorInfoOrigin = "Color from curvature";
+    }
+
+    onColorInfoOriginOrCurvColorChanged();
+}
+
+
+//*************************************************************************************************************
+
+void BrainSurfaceTreeItem::setVisible(bool state)
+{
+    m_pRenderable3DEntity->setParent(state ? m_pParentEntity : Q_NULLPTR);
+}
+
+
+//*************************************************************************************************************
+
 void BrainSurfaceTreeItem::onColorInfoOriginOrCurvColorChanged()
 {
     if(this->hasChildren()) {
-        QString sColorInfoOrigin = m_pItemSurfColorInfoOrigin->data(BrainTreeMetaItemRoles::SurfaceColorInfoOrigin).toString();
         QVariant data;
         QByteArray arrayNewVertColor;
 
-        if(sColorInfoOrigin == "Color from curvature") {
+        if(m_sColorInfoOrigin == "Color from curvature") {
```

Comment:
is this really a good comperator operation? how about m_sColorInfoOrigin.contains("curvature")?